app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java

```diff
@@ -0,0 +1,181 @@
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
+
+/**
+ * Tests for login screen, cannot be
+ */
+@RunWith(AndroidJUnit4.class)
+public class LoginActivityTest
+{
+    private Solo solo;
+
+    private Context appContext;
+    private Resources r;
+
+    private int shortWait = 1000;
+    private int longWait = 4000;
+
+    private EditText emailText;
+    private EditText passwordText;
+
+    @Rule
+    public ActivityTestRule<LoginActivity> rule =
+            new ActivityTestRule<>(LoginActivity.class, true, true);
+
+    /**
+     * Runs before all tests and creates solo instance.
+     * @throws Exception
+     */
+    @Before
+    public void setUp() throws Exception{
+        solo = new Solo(getInstrumentation(),rule.getActivity());
+
+        /* Gets context for app resources */
+        appContext = InstrumentationRegistry.getInstrumentation().getTargetContext();
+
+        /* Gets resource files */
+        r = appContext.getResources();
+
+        emailText = (EditText) solo.getView(R.id.login_email);
+        passwordText = (EditText) solo.getView(R.id.login_password);
+    }
+
+    /**
+     * Gets the activity
+     *
+     * @throws Exception
+     */
+    @Test
+    public void start() throws Exception {
+        Activity activity = rule.getActivity();
+    }
+
+    /**
+     * Tests going to CreateAccountActivity
+     */
+    @Test
+    public void createAccountButtonTest()
+    {
+        solo.clickOnButton(r.getString(R.string.create_account));
+        solo.assertCurrentActivity(r.getString(R.string.wrong_activity), CreateAccountActivity.class);
+
+        solo.sleep(shortWait);
+    }
+
+    /**
+     * Tests login button functionality with no input
+     */
+    @Test
+    public void emptyLoginTest()
+    {
+        solo.clickOnButton(r.getString(R.string.login));
+        solo.assertCurrentActivity(r.getString(R.string.wrong_activity), LoginActivity.class);
+
+        solo.sleep(shortWait);
+    }
+
+    /**
+     * Test for spaces input
+     */
+    @Test
+    public void spacesLoginTest()
+    {
+        solo.enterText(emailText, " ");
+        solo.enterText(passwordText, " ");
+        solo.clickOnButton(r.getString(R.string.login));
+
+        solo.sleep(shortWait);
+
+        Assert.assertNotNull(emailText.getError());
+        Assert.assertNotNull(passwordText.getError());
+    }
+
+    /**
+     * Tests for incorrect password error
+     */
+    @Test
+    public void wrongPasswordTest()
+    {
+        solo.enterText(emailText, r.getString(R.string.test_account1_email));
+        solo.enterText(passwordText, r.getString(R.string.wrong_pass));
+        solo.clickOnButton(r.getString(R.string.login));
+
+        solo.sleep(longWait);
+
+        Assert.assertNotNull(passwordText.getError());
```

Comment:
Same

Reply:
uhm, you cant getError().toString but you can check if text is visible

so if two edittexts both have Cannot be empty! i am only checking once if Cannot be empty exists

But if there are two different error messages showing at the same time,
can check if message1 exists
can check if message 2 exists

Reply:
updated